### ISSUE TITLE
Scenario Browser

### ISSUE THREAD

**Issue by sylae** [calref/cboe#22]
_Original author: sylaecorell (October 11, 2009 00:14:40)_

(reposting here, I think I picked a bad time to post this at SW)

OK, here's an idea that might be a little out there, but what if we could
add a 'scenario browser' to the custom scenario dialog? My idea is that,
when the user clicks the 'custom scenarios' button, there'll be a link at
the bottom of the dialog box that says 'Get more scenarios...' or some such.

When the user clicks on this, then are presented with a window containing a
table/listing of scenarios. This data would come from an XML file pulled
from a url (this url would be configurable in the ini file, so a user could
switch between 'The sub-official Community-updated database of scenarios'
or, for example, 'Celtic Minstrel's uber-awesome list of utility scenarios'
(or perhaps we could have an array of them stored, and users could pick the
used file from a drop-down in the dialog box)). My idea of the XML
structure is shown below. Anyway, the user could click a scenario to see
more info about it (author, version, a link to CSR, etc.) or be able to
sort and filter the choices. Then they pick one to download, and BoE gets
the archive of the scenario from the URL given by the XML file, unzips it
into the scenarios folder (you'd have to have a way to prevent overwrites
and the like), and (if there is a readme) gives a prompt to open the readme
file and/or start the scenario, or keep downloading scenarios.

_Original issue: http://code.google.com/p/openexile/issues/detail?id=20_

**Comment by CelticMinstrel** [comment]
This ends up being more or less equivalent in function to Wesnoth's add-on manager – which means we would need to be able to run a daemon on a remote host that the client can connect to. Though, depending on how it's implemented, the "daemon" could just be the regular web server.

**Comment by NQNStudios** [comment]
My plan for the Itch.io edition I'm building is to have the game search the Itch client's install directories for BoE scenarios. Then the official distribution technique will be to upload your scenario as a project on itch.io which can be installed through the client.

For extra credit, I might make a system where you can package your scenarios with an install script for the game itself, so each scenario on Itch can be installed and run standalone.

I think this will be much better than maintaining a daemon.

**Comment by CelticMinstrel** [comment]
> I think this will be much better than maintaining a daemon.

Except for the fact that it depends on the itch.io client. So, not really better than maintaining a daemon. It's not a bad idea, mind you, but it seems a bit questionable for that to be the sole distribution method.

**Comment by NQNStudios** [comment]
Right, the point of it from my perspective is not to be the one maintaining
the live service.
